src/Cryptol/TypeCheck/Infer.hs

```diff
@@ -1368,13 +1371,13 @@ checkExhaustive name as asmps guards =
   -- Try to validate all cases
   doGoals todo gs =
     case todo of
-      []     -> pure $ Just True
+      []     -> pure ProofSuccess
       alt : more ->
         do ok <- canProve (asmps ++ alt) gs
            case ok of
-             Just True -> doGoals more gs
-             Just False -> pure $ Just False
-             Nothing -> pure Nothing
+             ProofSuccess -> doGoals more gs
+             ProofFail -> pure ProofFail
+             ProofUnknown -> pure ProofFail
```

Comment:
This changed a `Nothing` value to `ProofFail`. Shouldn't this be `ProofUnknown`?

Reply:
You're right, fixed.

Reply:
My hope is that fixing this will also fix the test suite failures seen in the most recent CI pipeline. I'll approve this under the assumption that that is actually the case.